src/Plugin/Validation/Constraint/UniqueOgMembershipConstraintValidator.php

```diff
@@ -0,0 +1,62 @@
+<?php
+
+declare(strict_types = 1);
+
+namespace Drupal\og\Plugin\Validation\Constraint;
+
+use Symfony\Component\Validator\Constraint;
+use Symfony\Component\Validator\ConstraintValidator;
+
+/**
+ * Ensures that new members added to a group do not already exist.
+ *
+ * Note that in typical operation, this validation constraint will not come into
+ * play, as the membership entity's uid field is already validated by core's
+ * ValidReferenceConstraint, which hands over to the entity reference selection
+ * plugin. In our case, that is
+ * \Drupal\og\Plugin\EntityReferenceSelection\OgUserSelection, which already
+ * checks an existing member cannot be added to the group again.
+ */
+class UniqueOgMembershipConstraintValidator extends ConstraintValidator {
+
+  /**
+   * {@inheritdoc}
+   */
+  public function validate($value, Constraint $constraint) {
+    /* @var \Drupal\Core\Field\FieldItemInterface $value */
```

Comment:
Inline @var declarations should use the /** */ delimiters 

```suggestion
    /** @var \Drupal\Core\Field\FieldItemInterface $value */
```